src/VisualStudio/Core/Def/Implementation/PullMemberUp/MainDialog/PullMemberUpDialogViewModel.cs

```diff
@@ -0,0 +1,189 @@
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.PullMemberUp;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp.MainDialog
+{
+    internal class PullMemberUpDialogViewModel : AbstractNotifyPropertyChanged
+    {
+        public ImmutableArray<PullMemberUpSymbolViewModel> Members { get; set; }
+
+        public ImmutableArray<BaseTypeTreeNodeViewModel> Destinations { get; set; }
+
+        private BaseTypeTreeNodeViewModel _selectedDestination;
+
+        public BaseTypeTreeNodeViewModel SelectedDestination
+        {
+            get => _selectedDestination;
+            set
+            {
+                if (SetProperty(ref _selectedDestination, value, nameof(SelectedDestination)))
+                {
+                    var fields = Members.WhereAsArray(memberViewModel => memberViewModel.MemberSymbol.IsKind(SymbolKind.Field));
+                    if (_selectedDestination.MemberSymbol is INamedTypeSymbol interfaceSymbol &&
+                        interfaceSymbol.TypeKind == TypeKind.Interface)
+                    {
+                        // Disable field check box if destination is interface
+                        foreach (var member in fields)
+                        {
+                            member.IsCheckable = false;
+                        }
+                    }
+                    else
+                    {
+                        // Resume field check box back
+                        foreach (var member in fields)
+                        {
+                            member.IsCheckable = true;
+                        }
+                    }
+                    EnableOrDisableOkButton();
+                    CheckAndSetStateOfSelectAllCheckBox();
+                }
+            }
+        }
+
+        public ImmutableDictionary<ISymbol, Task<ImmutableArray<ISymbol>>> DependentsMap;
+
+        public ImmutableDictionary<ISymbol, PullMemberUpSymbolViewModel> SymbolToMemberViewMap { get; }
+
+        private bool _okButtonEnabled;
+
+        public bool OkButtonEnabled { get => _okButtonEnabled; set => SetProperty(ref _okButtonEnabled, value, nameof(OkButtonEnabled)); }
+
+        private bool? _selectAllCheckBoxState;
+
+        public bool? SelectAllCheckBoxState { get => _selectAllCheckBoxState; set => SetProperty(ref _selectAllCheckBoxState, value, nameof(SelectAllCheckBoxState)); }
+
+        private bool _threeStateEnable;
+
+        public bool ThreeStateEnable { get => _threeStateEnable; set => SetProperty(ref _threeStateEnable, value, nameof(ThreeStateEnable)); }
```

Comment:
Ahh, so it works in this way.
It controls the select all check box here. 
![image](https://user-images.githubusercontent.com/24360909/50321735-1cb17a00-0487-11e9-9018-48ade4917226.png)
So the behaviors are:
1.When some members are checked, it will be filled with a square(the third state). 
2.When no members are checked, it will be blank.
3.When all members are checked, it will be checked.

4.When it is in third state, click it will cause it to select all. 
5.When it is checked, click it will clear all selection. 
6.When it is unchecked, click it will cause select all.

I implement it in this way. 
When all members are checked, or unchecked, three state is disabled (it is a property of wpf checkbox), so it match behavior 5, 6.

When some members are checked, three state is enable. Select more will cause a helper function called 'CheckAndSetStateOfSelectAllCheckBox' to check the state. It will satisfy behavior 1, 2, 3.

And Finally, for behavior 4, that's the reason I create an my own checkbox in this PR. Since the default behavior of wpf standard checkbox is 
null(third state) -> false(not checked) -> true(checked), therefore I reverse this order to meet behavior 4. null (third state) -> true (checked). If it is true, then three state is disabled, go back to behavior 5, 6.

It is a little bit complicated.  Maybe I should put comments somewhere.